src/OTSubscriber.js

```diff
@@ -40,9 +40,19 @@ export default class OTSubscriber extends Component {
     const { streamProperties } = this.props;
     if (!isEqual(this.state.streamProperties, streamProperties)) {
       each(streamProperties, (individualStreamProperties, streamId) => {
-        const { subscribeToAudio, subscribeToVideo } = individualStreamProperties;
-        OT.subscribeToAudio(streamId, subscribeToAudio);
-        OT.subscribeToVideo(streamId, subscribeToVideo);
+        const { subscribeToAudio, subscribeToVideo, preferredResolution, preferredFrameRate } = individualStreamProperties;
+        if (subscribeToAudio !== undefined) {
```

Comment:
Why are we adding this check here? The sanitization should take care of this:
https://github.com/opentok/opentok-react-native/blob/master/src/helpers/OTHelper.js#L33

Reply:
The if clause is needed for the following case. Let's say I have the following case: 

```
const streamProperties = {
  ...this.state.streamProperties,
  [streamId]: {
    subscribeToAudio: false,
    subscribeToVideo: false,
    preferredResolution: {width: 320, height: 240}
  },
};

<OTSubscriber
  eventHandlers={this.subscriberEventHandlers}
  streamProperties={this.state.streamProperties}
></OTSubscriber>
```

If I update the `preferredResolution` prop of `streamProperties`, on the OTSubscriber didUpdate [hook](https://github.com/enricop89/opentok-react-native/blob/add-preferred-res-framerate/src/OTSubscriber.js#L43), the only property that has changed is the `preferredResolution` indeed. This means that `subscriberToAudio` and `subscribeToVideo` are undefined at this [point](https://github.com/enricop89/opentok-react-native/blob/add-preferred-res-framerate/src/OTSubscriber.js#L43). If undefined is passed into [OTHelper-SanitizeBoolean](https://github.com/opentok/opentok-react-native/blob/master/src/helpers/OTHelper.js#L33), it returns true, which is not wanted.

I hope it's clear 😃